sw/device/tests/rv_timer_test.c

```diff
@@ -32,28 +43,58 @@ bool test_main(void) {
 
   irq_global_ctrl(true);
   irq_timer_ctrl(true);
-  rv_timer_intr_enable(hart, true);
 
-  intr_handling_success = false;
-  rv_timer_set_us_tick(hart);
-  rv_timer_set_cmp(hart, cmp);
-  rv_timer_ctrl(hart, true);
+  mmio_region_t timer_reg =
+      mmio_region_from_addr(TOP_EARLGREY_RV_TIMER_BASE_ADDR);
+  CHECKZ(dif_rv_timer_init(
+      timer_reg,
+      (dif_rv_timer_config_t){.hart_count = 1, .comparator_count = 1}, &timer));
+
+  CHECKZ(
+      dif_rv_timer_irq_enable(&timer, kHart, kComparator, kDifRvTimerEnabled));
+
+  dif_rv_timer_tick_params_t tick_params;
+  CHECKZ(dif_rv_timer_approximate_tick_params(kClockFreqHz, kTickFreqHz,
+                                              &tick_params));
+  CHECKZ(dif_rv_timer_set_tick_params(&timer, kHart, tick_params));
+
+  uint64_t current_time;
+  CHECKZ(dif_rv_timer_counter_read(&timer, kHart, &current_time));
+  LOG_INFO("Current time: %d; timer theshold: %d", (uint32_t)current_time,
+           (uint32_t)(current_time + kDeadline));
+  CHECKZ(
+      dif_rv_timer_arm(&timer, kHart, kComparator, current_time + kDeadline));
 
-  CHECKZ(dif_gpio_all_write(&gpio, 0xFF00));  // all LEDs on
+  irq_fired = false;
+  CHECKZ(dif_rv_timer_counter_set_enabled(&timer, kHart, kDifRvTimerEnabled));
 
-  while (!intr_handling_success) {
+  CHECKZ(dif_gpio_all_write(&gpio, 0xFF00));  // All LEDs on.
```

Comment:
This is a sanity test that is intended to quickly test the timer functionality (will be run in CI verilator simulation). I would say that unless other peripherals are absolutely required in this test, it would be a good idea to minimise dependencies on other peripherals. In this case GPIO blinking (If I understand correctly), is not paramount to functioning of this test, but only to provide a visual confirmation to a user when running on FPGA? Using GPIOs pulls in two extra peripherals: GPIO and Pinmux, which I think can and should be avoided.

Reply:
This is a good point. There's a place for an "example" which shows setting up the timer and blinking an LED. That's separate from this which should actually test the timer alone (as much as possible).

Reply:
I deleted this. It was leftover from the original rv_timer_test.c that the sanity test is based off of.